services/server.go

```diff
@@ -90,41 +89,28 @@ func (ss *Server) Update(model *models.Server) error {
 }
 
 func (ss *Server) Delete(id string) error {
-	//if we are already in a transaction, use the existing transaction
-	inTrans := false
-	var trans *gorm.DB
-	if _, ok := ss.DB.CommonDB().(*sql.Tx); ok {
-		trans = ss.DB
-		inTrans = true
-	} else {
-		trans = ss.DB.Begin()
-		defer trans.RollbackUnlessCommitted()
-	}
-
-	model := &models.Server{
-		Identifier: id,
-	}
-
-	err := trans.Delete(models.Permissions{}, "server_identifier = ?", id).Error
-	if err != nil {
-		return err
-	}
-
-	err = trans.Delete(models.Client{}, "server_id = ?", id).Error
-	if err != nil {
-		return err
-	}
-
-	err = ss.DB.Delete(model).Error
-	if err != nil {
-		return err
-	}
-
-	if inTrans {
+	trans := ss.DB
```

Comment:
This call is interesting, because it actually looks to see if we are in a transaction already, and not create a new one if we are. We should keep this logic, unless the delete does the daemon call first.
This is because if the daemon doesn't delete the server, we have to roll the database back. This would always create a new one and commit it even if the daemon fails.